subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2094,9 +2094,70 @@ void lwm2m_engine_get_queue_mode(char *queue)
 	}
 }
 
+static int lwm2m_engine_allocate_resource_instance(struct lwm2m_engine_res *res,
+						   struct lwm2m_engine_res_inst **res_inst,
+						   uint8_t resource_instance_id)
+{
+	int i;
+
+	if (!res->res_instances || res->res_inst_count == 0) {
+		return -ENOMEM;
+	}
+
+	for (i = 0; i < res->res_inst_count; i++) {
+		if (res->res_instances[i].res_inst_id ==
+		    RES_INSTANCE_NOT_CREATED) {
+			break;
+		}
+	}
+
+	if (i >= res->res_inst_count) {
+		return -ENOMEM;
+	}
+
+	res->res_instances[i].res_inst_id = resource_instance_id;
+	*res_inst = &res->res_instances[i];
+	return 0;
+}
+
+int lwm2m_engine_get_create_res_inst(struct lwm2m_obj_path *path, struct lwm2m_engine_res **res,
+				     struct lwm2m_engine_res_inst **res_inst)
+{
+	int ret;
+	struct lwm2m_engine_res *r = NULL;
+	struct lwm2m_engine_res_inst *r_i = NULL;
```

Comment:
Drop the NULL initialization, because `path_to_objs()` should initialize these, unless there are some errors.
By having them NULL, you might hide so problem.

Reply:
At path_to_objs()
if (ri && res_inst) {
		*res_inst = ri;
	} 
that have to be changed to:
if (res_inst) {
		*res_inst = ri;
	}
	
	If I remove NULL Initialize

Reply:
Ah, I see.
I think it will make sense to refactor it to
```c
	if (res_inst) {
		*res_inst = ri;
	}
```

because all other members are stored. So if resource instance is not found, we will still write NULL into it.

Reply:
That check validation have been added 3 years ago. I need to check That if we set to NULL it not break anything.